src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -132,21 +146,32 @@ public class ProvenBlockHeaderStore : IProvenBlockHeaderStore
             this.asyncLoopFactory = asyncLoopFactory;
 
             this.performanceCounter = new BackendPerformanceCounter(dateTimeProvider);
-            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark, 100);
+            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark);
+            nodeStats.RegisterStats(this.AddComponentStats, StatsType.Component);
         }
 
         /// <inheritdoc />
-        public async Task InitializeAsync()
+        public async Task InitializeAsync(ChainedHeader chainedHeader)
```

Comment:
Comment on what is this chained header and where it should come from is required. 
In this review I'm assuming it's the consensus tip right after CM initialization

Reply:
Ok.